coding/logging.rst

```diff
@@ -0,0 +1,231 @@
+   traceLogger.debug("On %d-th iteration of star selection", iteration)
+   innerTraceLogger = lsst.log.getLogger("TRACE2.meas.algorithms.starSelector.catalogReader")
+   innerTraceLogger.debug("Reading catalog %s", catalogName)
+   # Or log to a component directly
+   lsst.log.log("TRACE4.meas.algorithms.starSelector.psfCandidate", lsst.log.DEBUG, "Making a psfCandidate from star %d", starId)
+
+Notice that all loggers in the hierarchy under a given component at a given trace level can be enabled easily using, e.g., ``TRACE2.lsst.meas.algorithms.starSelector``.
+Besides, a utility function :lfunc:`lsst.log.utils.traceSetAt()` is provided to adjust logging level of a group of loggers so to display messages with trace number <= NUMBER. This is demostrated in the following example:
+
+.. literalinclude:: logging_snippets/tracing.py
+   :language: python
+
+The example can be run if :lmod:`lsst.log` is setup:
+
+.. code-block:: shell
+
+   $ python logging_snippets/tracing.py
+   INFO  root: Setting trace at 0
+   INFO  root: Writing 6 debug messages
+   DEBUG TRACE0.example.component: Fine tracing to TRACE0
+   INFO  root: Setting trace at 1
+   INFO  root: Writing 6 debug messages
+   DEBUG TRACE0.example.component: Fine tracing to TRACE0
+   DEBUG TRACE1.example.component: Fine tracing to TRACE1
+   INFO  root: Setting trace at 2
+   INFO  root: Writing 6 debug messages
+   DEBUG TRACE0.example.component: Fine tracing to TRACE0
+   DEBUG TRACE1.example.component: Fine tracing to TRACE1
+   DEBUG TRACE2.example.component: Fine tracing to TRACE2
+   INFO  root: Setting trace at 3
+   INFO  root: Writing 6 debug messages
+   DEBUG TRACE0.example.component: Fine tracing to TRACE0
+   DEBUG TRACE1.example.component: Fine tracing to TRACE1
+   DEBUG TRACE2.example.component: Fine tracing to TRACE2
+   DEBUG TRACE3.example.component: Fine tracing to TRACE3
+   INFO  root: Setting trace at 4
+   INFO  root: Writing 6 debug messages
+   DEBUG TRACE0.example.component: Fine tracing to TRACE0
+   DEBUG TRACE1.example.component: Fine tracing to TRACE1
+   DEBUG TRACE2.example.component: Fine tracing to TRACE2
+   DEBUG TRACE3.example.component: Fine tracing to TRACE3
+   DEBUG TRACE4.example.component: Fine tracing to TRACE4
+   INFO  root: Setting trace at 5
+   INFO  root: Writing 6 debug messages
+   DEBUG TRACE0.example.component: Fine tracing to TRACE0
+   DEBUG TRACE1.example.component: Fine tracing to TRACE1
+   DEBUG TRACE2.example.component: Fine tracing to TRACE2
+   DEBUG TRACE3.example.component: Fine tracing to TRACE3
+   DEBUG TRACE4.example.component: Fine tracing to TRACE4
+   DEBUG TRACE5.example.component: Fine tracing to TRACE5
+
+
+Porting from pex_logging
+========================
+
+Logging in the science pipeline codes is being migrated from using :lmod:`lsst.pex.logging` to :lmod:`lsst.log`.
+Below some examples are listed and may be used as a starting point of porting:
```

Comment:
- `Below,`
- `and may be used` → `that may be used`
- `of porting` → `for porting`